DFReader.py

```diff
@@ -755,19 +755,28 @@ def init_arrays(self, progress_callback=None):
         HEAD2 = self.HEAD2
         lengths = [-1] * 256
 
+        if self.data_len < 3:
+            raise Exception("Cannot parse, file has length %d" % self.data_len)
+
         while ofs+3 < self.data_len:
             hdr = self.data_map[ofs:ofs+3]
             if hdr[0] != HEAD1 or hdr[1] != HEAD2:
-                print("bad header 0x%02x 0x%02x" % (u_ord(hdr[0]), u_ord(hdr[1])), file=sys.stderr)
-                ofs += 1
-                continue
+                # avoid end of file garbage, block based logs can have up to 256 bytes of zeros
+                if self.data_len - ofs >= 528 or self.data_len < 528:
+                    print("bad header 0x%02x 0x%02x at %d" % (u_ord(hdr[0]), u_ord(hdr[1]), ofs), file=sys.stderr)
+                    ofs += 1
```

Comment:
This seems to be changing logic, not just squelching warnings.

Reply:
Fixed